kodi_addon_checker/check_string.py

```diff
@@ -38,3 +39,54 @@ def find_blacklisted_strings(report: Report, addon_path: str, problems: list, wa
     for result in handle_files.find_in_file(addon_path, warnings, file_types):
         report.add(Record(WARNING, "Found blacklisted term %s in file %s:%s (%s)"
                           % (result["term"], result["searchfile"], result["linenumber"], result["line"])))
+
+
+def check_for_invalid_strings_po(report: Report, file_index: list):
+    """Validate strings.po files
+        :file_index: list having names and path of all the files present in addon
+    """
+    po_file_index = [f for f in file_index if f["name"] == "strings.po"]
+    report_made = False
+
+    for po_file in po_file_index:
+        full_path = os.path.join(po_file["path"], po_file["name"])
+
+        with open(full_path, "r", encoding="utf-8") as f:
+            contents = f.read()
+
+        if not contents:
+            report_made = True
+            report.add(Record(PROBLEM, "Invalid PO file %s: File is empty" % (relative_path(full_path))))
+            continue
+
+        if "\r\n" in contents:
+            report_made = True
+            report.add(Record(WARNING, "Windows line endings found in %s, consider converting to Linux line endings."
+                              % relative_path(full_path)))
+
```

Comment:
Add the following BOM check here, otherwise `polib.pofile(...)` will report `Syntax error on line 1`, which isn't very clear.
```suggestion

        if contents[0] == '\ufeff':
            report_made = True
            report.add(Record(PROBLEM, "Invalid PO file %s: File contains BOM (byte order mark)" % (relative_path(full_path))))
            continue
```

Reply:
I added this after the required header check, and before `polib.pofile(...)`